hal/armv7r/zynqmp/ddr_init.c

```diff
@@ -0,0 +1,433 @@
+	{ ddrc_addrmap3, 0x0f0f0f0f, 0x01010101 },
+	{ ddrc_addrmap4, 0x00000f0f, 0x00000f0f },
+	{ ddrc_addrmap5, 0x0f0f0f0f, 0x070f0707 },
+	{ ddrc_addrmap6, 0x8f0f0f0f, DDRC_ADDRMAP6_VAL },
+	{ ddrc_addrmap7, 0x00000f0f, DDRC_ADDRMAP7_VAL },
+	{ ddrc_addrmap8, 0x00001f1f, DDRC_ADDRMAP8_VAL },
+	{ ddrc_addrmap9, 0x0f0f0f0f, 0x07070707 },
+	{ ddrc_addrmap10, 0x0f0f0f0f, 0x07070707 },
+	{ ddrc_addrmap11, 0x0000000f, 0x00000007 },
+	{ ddrc_odtcfg, 0x0f1f0f7c, 0x06000600 },
+	{ ddrc_odtmap, 0x00003333, 0x00000001 },
+	{ ddrc_sched, 0x7fff3f07, 0x01002001 },
+	{ ddrc_perflpr1, 0xff00ffff, 0x08000040 },
+	{ ddrc_perfwr1, 0xff00ffff, DDRC_PERFWR1_VAL },
+	{ ddrc_dqmap0, 0xffffffff, 0x00000000 },
+	{ ddrc_dqmap1, 0xffffffff, 0x00000000 },
+	{ ddrc_dqmap2, 0xffffffff, 0x00000000 },
+	{ ddrc_dqmap3, 0xffffffff, 0x00000000 },
+	{ ddrc_dqmap4, 0x0000ffff, 0x00000000 },
+	{ ddrc_dqmap5, 0x00000001, 0x00000001 },
+	{ ddrc_dbg0, 0x00000011, 0x00000000 },
+	{ ddrc_dbgcmd, 0x80000033, 0x00000000 },
+	{ ddrc_swctl, 0x00000001, 0x00000000 },
+	{ ddrc_pccfg, 0x00000111, 0x00000001 },
+	{ ddrc_pcfgr_0, 0x000073ff, 0x0000200f },
+	{ ddrc_pcfgw_0, 0x000073ff, 0x0000200f },
+	{ ddrc_pctrl_0, 0x00000001, 0x00000001 },
+	{ ddrc_pcfgqos0_0, 0x0033000f, 0x0020000b },
+	{ ddrc_pcfgqos1_0, 0x07ff07ff, 0x00000000 },
+	{ ddrc_pcfgr_1, 0x000073ff, 0x0000200f },
+	{ ddrc_pcfgw_1, 0x000073ff, 0x0000200f },
+	{ ddrc_pctrl_1, 0x00000001, 0x00000001 },
+	{ ddrc_pcfgqos0_1, 0x03330f0f, 0x02000b03 },
+	{ ddrc_pcfgqos1_1, 0x07ff07ff, 0x00000000 },
+	{ ddrc_pcfgr_2, 0x000073ff, 0x0000200f },
+	{ ddrc_pcfgw_2, 0x000073ff, 0x0000200f },
+	{ ddrc_pctrl_2, 0x00000001, 0x00000001 },
+	{ ddrc_pcfgqos0_2, 0x03330f0f, 0x02000b03 },
+	{ ddrc_pcfgqos1_2, 0x07ff07ff, 0x00000000 },
+	{ ddrc_pcfgr_3, 0x000073ff, 0x0000200f },
+	{ ddrc_pcfgw_3, 0x000073ff, 0x0000200f },
+	{ ddrc_pctrl_3, 0x00000001, 0x00000001 },
+	{ ddrc_pcfgqos0_3, 0x0033000f, 0x00100003 },
+	{ ddrc_pcfgqos1_3, 0x07ff07ff, 0x0000004f },
+	{ ddrc_pcfgwqos0_3, 0x0033000f, 0x00100003 },
+	{ ddrc_pcfgwqos1_3, 0x000007ff, 0x0000004f },
+	{ ddrc_pcfgr_4, 0x000073ff, 0x0000200f },
+	{ ddrc_pcfgw_4, 0x000073ff, 0x0000200f },
+	{ ddrc_pctrl_4, 0x00000001, 0x00000001 },
+	{ ddrc_pcfgqos0_4, 0x0033000f, 0x00100003 },
+	{ ddrc_pcfgqos1_4, 0x07ff07ff, 0x0000004f },
+	{ ddrc_pcfgwqos0_4, 0x0033000f, 0x00100003 },
+	{ ddrc_pcfgwqos1_4, 0x000007ff, 0x0000004f },
+	{ ddrc_pcfgr_5, 0x000073ff, 0x0000200f },
+	{ ddrc_pcfgw_5, 0x000073ff, 0x0000200f },
+	{ ddrc_pctrl_5, 0x00000001, 0x00000001 },
+	{ ddrc_pcfgqos0_5, 0x0033000f, 0x00100003 },
+	{ ddrc_pcfgqos1_5, 0x07ff07ff, 0x0000004f },
+	{ ddrc_pcfgwqos0_5, 0x0033000f, 0x00100003 },
+	{ ddrc_pcfgwqos1_5, 0x000007ff, 0x0000004f },
+	{ ddrc_sarbase0, 0x000001ff, 0x00000000 },
+	{ ddrc_sarsize0, 0x000000ff, 0x00000000 },
+	{ ddrc_sarbase1, 0x000001ff, 0x00000010 },
+	{ ddrc_sarsize1, 0x000000ff, 0x0000000f },
+	{ ddrc_dfitmg0_shadow, 0x1fbfbf3f, 0x07828002 },
+};
+
+
+static struct {
```

Comment:
otherwise will be copied to .data
```suggestion
static const struct {
```